warehouse/query-core/src/main/java/datawave/query/tracking/ActiveQueryLog.java

```diff
@@ -0,0 +1,198 @@
+package datawave.query.tracking;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.apache.accumulo.core.conf.AccumuloConfiguration;
+import org.apache.accumulo.core.data.Range;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+
+public class ActiveQueryLog {
+    
+    private static Logger LOG = LoggerFactory.getLogger(ActiveQueryLog.class);
+    private static ActiveQueryLog instance = null;
+    private static AccumuloConfiguration conf = null;
+    
+    // Accumulo properties
+    public static final String MAX_IDLE = "datawave.query.active.maxIdleMs";
+    public static final String LOG_PERIOD = "datawave.query.active.logPeriodMs";
+    public static final String LOG_MAX_QUERIES = "datawave.query.active.logMaxQueries";
+    public static final String WINDOW_SIZE = "datawave.query.active.windowSize";
+    
+    // Changeable via Accumulo properties
+    private long maxIdle = 900000;
+    private long logPeriod = 60000;
+    private int logMaxQueries = 5;
+    private int windowSize = 10;
+    
+    private Cache<String,ActiveQuery> CACHE = null;
```

Comment:
Given it appears that setMaxIdle could never be called, then the CACHE could remain null and then the get() methods would throw an NPE.  I think we should give it an initial value here to guarentee that cannot happen.

Reply:
The private constructor (called from getInstance()) of ActiveQueryLog calls either 
setMaxIdle directly
or
checkSettings [with useDefaults=true] (which calls setMaxIdle with either the default value of the value from conf)

Reply:
What if the conf does not have a value for maxIdle?